enhancements/ingress/add-dns-and-loadbalancer-conditions-to-managed-gateway.md

```diff
@@ -0,0 +1,624 @@
+additional conditions.
+
+**Risk: Exposure of sensitive information in condition messages**
+* Mitigation: Sanitize error messages to remove any credentials or sensitive
+data before including in conditions
+
+**Risk: Compatibility with future Gateway API versions**
+* Mitigation: Use well-defined condition types that align with Kubernetes
+conventions and Gateway API patterns
+
+**Security Review:**
+* No new authentication or authorization mechanisms introduced
+* Conditions are read-only status information
+* Standard RBAC applies to Gateway resources
+* Cluster Ingress Operator cluster role needs a new permission to `UPDATE` and `PATCH`
+Gateway status subresource.
+
+**UX Review:**
+* Condition messages should be clear and actionable for both administrators and developers
+* Error messages should include enough detail to enable self-service troubleshooting
+
+### Drawbacks
+
+**Additional API surface:**
+* Adding conditions increases the Gateway status size, though the impact is
+minimal (~1KB per Gateway)
+
+**Maintenance burden:**
+* Requires ongoing maintenance to keep condition logic in sync with DNS operator and cloud provider behavior
+* May need updates when DNS operator or cloud provider integrations change
+
+**Potential confusion:**
+* Users may expect these conditions to appear on all Gateway resources, not just
+those in `openshift-ingress`
+* Clear documentation will be needed to explain the scoping
+
+**Not applicable to all environments:**
+* The LoadBalancer condition is only meaningful on cloud platforms or platforms with
+`LoadBalancer` support.
+* Users on bare metal may see persistent `False` or `Unknown` status which could
+be confusing
+
+## Alternatives (Not Implemented)
+
+**Alternative 1: Use Gateway API standard conditions only**
+* The Gateway API specification defines standard conditions like `Accepted` and `Programmed`
+* We could map DNS/LB status into these existing conditions rather than adding new ones
+* Rejected because: Standard conditions don't provide the granularity needed to
+distinguish DNS vs. LoadBalancer issues
+
+**Alternative 2: Add conditions to all Gateway resources cluster-wide**
+* Could add these conditions to any Gateway resource, not just those in `openshift-ingress`
+* Rejected because: User-managed Gateways may not follow the same DNS/LB
+provisioning patterns, and this would create inconsistent behavior
+
+**Alternative 3: Create separate status CRD for infrastructure details**
+* Could create a new CRD (e.g., `GatewayInfrastructureStatus`) to hold DNS/LB information
+* Rejected because: Adds unnecessary complexity; conditions are the standard
+Kubernetes pattern for status reporting
+
+**Alternative 4: Emit events instead of conditions**
+* Could use Kubernetes events to report DNS/LB status instead of conditions
+* Rejected because: Events are transient and harder to query programmatically;
+conditions provide persistent, queryable status
+
+## Open Questions [optional]
```

Comment:
What about conflicting hostnames? With the current DNS implementation from CIO, DNS resource record will have multiple entries for the same hostname round robin between gateway controllers.

Reply:
> With the current DNS implementation from CIO, DNS resource record will have multiple entries for the same hostname round robin between gateway controllers.

We only publish a DNS record for one of two conflicting gateways.  Unfortunately, the exact behavior varies by platform: on AWS, the second wins; on GCP, the first wins.  I don't know what happens on Azure, IBM Cloud, or other platforms.  That's all the more reason we need https://github.com/openshift/cluster-ingress-operator/pull/1229.